platform/iphone/in_app_store.mm

```diff
@@ -116,167 +92,150 @@ - (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProdu
 	PoolStringArray invalid_ids;
 
 	for (NSString *ipid in response.invalidProductIdentifiers) {
-
 		invalid_ids.push_back(String::utf8([ipid UTF8String]));
-	};
+	}
 	ret["invalid_ids"] = invalid_ids;
 
 	InAppStore::get_singleton()->_post_event(ret);
+}
 
-	[request release];
-};
-
-@end
-
-Error InAppStore::request_product_info(Variant p_params) {
-
-	Dictionary params = p_params;
-	ERR_FAIL_COND_V(!params.has("product_ids"), ERR_INVALID_PARAMETER);
-
-	PoolStringArray pids = params["product_ids"];
-	printf("************ request product info! %i\n", pids.size());
-
-	NSMutableArray *array = [[[NSMutableArray alloc] initWithCapacity:pids.size()] autorelease];
-	for (int i = 0; i < pids.size(); i++) {
-		printf("******** adding %ls to product list\n", pids[i].c_str());
-		NSString *pid = [[[NSString alloc] initWithUTF8String:pids[i].utf8().get_data()] autorelease];
-		[array addObject:pid];
-	};
-
-	NSSet *products = [[[NSSet alloc] initWithArray:array] autorelease];
-	SKProductsRequest *request = [[SKProductsRequest alloc] initWithProductIdentifiers:products];
-
-	ProductsDelegate *delegate = [[ProductsDelegate alloc] init];
-
-	request.delegate = delegate;
-	[request start];
-
-	return OK;
-};
-
-Error InAppStore::restore_purchases() {
-
-	printf("restoring purchases!\n");
-	[[SKPaymentQueue defaultQueue] restoreCompletedTransactions];
-
-	return OK;
-};
-
-@interface TransObserver : NSObject <SKPaymentTransactionObserver> {
-};
-@end
-
-@implementation TransObserver
-
+#pragma mark - SKPaymentTransactionObserver
 - (void)paymentQueue:(SKPaymentQueue *)queue updatedTransactions:(NSArray *)transactions {
-
-	printf("transactions updated!\n");
 	for (SKPaymentTransaction *transaction in transactions) {
+		// Create a variant with this info
+		Dictionary ret = TransactionDictionary(transaction);
+		String pid = String::utf8([transaction.payment.productIdentifier UTF8String]);
```

Comment:
Some cases need the `pid`, so we'll avoid parsing it individually for those cases.